tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.out

```diff
@@ -0,0 +1,18 @@
+Error found:
+in module [33mMain[0m
+at tests/purs/failing/ProgrammablePolykindedTypeErrorsTypeString.purs:23:7 - 23:17 (line 23, column 7 - line 23, column 17)
+
+  A custom type error occurred while solving type class constraints:
+
+    Don't want to show Just @Type String because.
```

Comment:
I think we should probably strip the kind annotations out here (the `@Type`) bit.

Reply:
We still want to show them when `--verbose-errors` is set, right?

Reply:
If we already decided on a strategy of showing kind annotations when verbose errors is on, then applying that same strategy here sounds good. I can’t remember whether that should be happening anyway; do the tests run with verbose errors on?

Reply:
As far as I can tell, in the only places we're currently erasing kind annotations in error messages, we do so conditional on verbose errors being off. I believe the tests don't run with verbose errors, which maybe makes testing that the erasure is conditional on verbosity difficult.

Also, it seems there are plenty of other places where we don't do this and perhaps should (just grep for `@` in `tests/**/*.out`), which maybe makes dumping this request on this PR a little unfair? I agree that it should be done, but maybe we should clean this up holistically (and figure out how to test it robustly) in a separate PR.

Reply:
Ok, yes, that makes sense.

Reply:
We don’t strip them in the error “trace” to aid debugging.